nntrainer/layers/tensor_layer.cpp

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2025 Jijoong Moon <jijoong.moon@samsung.com>
+ *
+ * @file   tensor_layer.cpp
+ * @date   17 Jan 2025
+ * @brief  This is QNN Tensor Layer Class of Neural Network
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Jijoong Moon <jijoong.moon@samsung.com>
+ * @bug    No known bugs except for NYI items
+ *
+ */
+
+#include <iostream>
+#include <layer_context.h>
+#include <nntrainer_error.h>
+#include <nntrainer_log.h>
+#include <node_exporter.h>
+#include <string>
+#include <tensor_layer.h>
+#include <tensor_wrap_specs.h>
+#include <util_func.h>
+
+namespace nntrainer {
+
+static constexpr size_t SINGLE_INOUT_IDX = 0;
+
+TensorLayer::TensorLayer() : Layer(), tensor_props({}, {}, {}, {}) {}
+
+void TensorLayer::setProperty(const std::vector<std::string> &values) {
+  auto remain_props = loadProperties(values, tensor_props);
+  NNTR_THROW_IF(!remain_props.empty(), std::invalid_argument)
+    << "[TensorLayer] Unknown Layer Properties count " +
+         std::to_string(values.size());
+}
+
+void TensorLayer::finalize(InitLayerContext &context) {
+  auto &dims = std::get<std::vector<props::TensorDimension>>(tensor_props);
+  std::vector<TensorDim> t_dims(dims.begin(), dims.end());
+
+  auto &t_dtype = std::get<std::vector<props::TensorDataType>>(tensor_props);
+  auto &t_name = std::get<std::vector<props::TensorName>>(tensor_props);
+  auto &t_life = std::get<std::vector<props::TensorLife>>(tensor_props);
+
+  NNTR_THROW_IF(!t_dims.size(), std::invalid_argument)
+    << "Tensor dimension is not provided";
+  n_tensor = t_dims.size();
+
+  if (!t_dtype.size()) {
+    ml_logi("Set Activation Tensor DataType");
+    t_dtype.reserve(t_dims.size());
+    for (auto t : t_dims)
+      t_dtype.push_back(context.getActivationDataType());
+  }
+
+  if (!t_life.size()) {
+    ml_logi("Set max Tensor LifeSpan");
+    t_life.reserve(t_dims.size());
+    for (auto t : t_dims)
+      t_life.push_back(nntrainer::TensorLifespan::MAX_LIFESPAN);
+  }
+
+  auto engine = context.getComputeEngineType();
+
+  NNTR_THROW_IF(t_dims.size() != t_dtype.size(),
+                std::invalid_argument)
+    << "Size of Dimensions, Types, Formats should be matched!";
```

Comment:
It seems, `InitLayerContext provides singular format provided by `InitLayerContext::getFormat()` for shapes (t_dims[i]) line 72;

Should the Format be mentioned here then?